integration/tests/neon_evm/test_revision.py

```diff
@@ -0,0 +1,314 @@
+
+
+def test_1_user_send_2_parallel_trx_with_neon_balance_change(operator_keypair, treasury_pool, neon_api_client,
+                                                             session_user, rw_lock_contract,
+                                                             user_account, evm_loader, holder_acc, new_holder_acc):
+    amount = 1000000
+    deposit_neon(evm_loader, operator_keypair, session_user.eth_address, 3 * amount)
+    recipients = [make_new_user(evm_loader), make_new_user(evm_loader)]
+    contract = deploy_contract(operator_keypair, session_user, "transfers", evm_loader, treasury_pool)
+
+    sender_revisions_before = evm_loader.get_balance_account_revision(session_user.balance_account_address)
+    recipients_eth_addresses = [rec.eth_address for rec in recipients]
+    signed_tx1 = make_contract_call_trx(session_user, contract, 'transferNeon(uint256,address[])',
+                                        [amount, recipients_eth_addresses],
+                                        value=3 * amount)
+    accounts = [rec.balance_account_address for rec in recipients] + \
+               [rec.solana_account_address for rec in recipients]
+    accounts += [session_user.balance_account_address,
+                 session_user.solana_account_address,
+                 contract.balance_account_address,
+                 contract.solana_address]
+
+    write_transaction_to_holder_account(signed_tx1, holder_acc, operator_keypair)
+    send_transaction_step_from_account(operator_keypair, evm_loader, treasury_pool, holder_acc,
+                                       accounts, EVM_STEPS, operator_keypair)
+    send_transaction_step_from_account(operator_keypair, evm_loader, treasury_pool, holder_acc,
+                                       accounts, EVM_STEPS, operator_keypair)
+
+    signed_tx2 = make_contract_call_trx(session_user, contract, 'transferNeon(uint256,address[])',
+                                        [amount, recipients_eth_addresses],
+                                        value=3 * amount)
+
+    resp = execute_trx_from_instruction(operator_keypair, evm_loader, treasury_pool.account, treasury_pool.buffer,
+                                        signed_tx2,
+                                        accounts,
+                                        operator_keypair)
+    check_transaction_logs_have_text(resp.value, "exit_status=0x11")
+
+    send_transaction_step_from_account(operator_keypair, evm_loader, treasury_pool, holder_acc,
+                                       accounts, EVM_STEPS, operator_keypair)
+    resp = send_transaction_step_from_account(operator_keypair, evm_loader, treasury_pool, holder_acc,
+                                              accounts, EVM_STEPS, operator_keypair)
+    check_transaction_logs_have_text(resp.value.transaction.transaction.signatures[0], "exit_status=0x11")
+    check_holder_account_tag(holder_acc, FINALIZED_STORAGE_ACCOUNT_INFO_LAYOUT, TAG_FINALIZED_STATE)
```

Comment:
could we understand that some transaction started from the beginning because revision was changed? If yes, it will be a good check

Reply:
It should be 1, because each user changes his own storage cells
I can add this check but I think it is the same as in test_call_contract_with_changing_data test

Reply:
No, it's imposible 
we can get which revisions holder account remembers but it's not easy